Radar rewrite

chrisc06 opened the pull request:
Rewritten most of the logic
Improved initial design
Made it mobile friendly (touch, scaling)
Added Zooming, moving around
Fixed ancient positioning
Added a connection status
You can see teammates now
Modified config with teammate toggle, enemy color (hp based or just a color) Added improved error checking to the typescript
Improved update_data

chrisc06 committed:
Radar rewrite

Rewritten most of the logic
Improved initial design
Made it mobile friendly (touch, scaling)
Added Zooming, moving around
Fixed ancient positioning
Added a connection status
You can see teammates now
Modified config with teammate toggle, enemy color (hp based or just a color)
Added improved error checking to the typescript
Improved update_data

Sonys9 commented:
goood

Dispnser commented:
LGTM

avitran0 commented:
looks mostly good. i would put the radar configuration not in the cheat (that should *only send the data*), but in the web interface. the settings should be saved/loaded with localStorage, and thus can be configured per-user.

chrisc06 commented:
Got it, give me a moment, im also adding viewer count for fun

chrisc06 committed:
Radar changes

Made savavble settings an option in the webserver instead of it being part in the client gui
Removed all config related stuff from the client
Fixed websocket reconnect spamming instead of waiting 5secs
Added a live viewer count

avitran0 commented:
looks fine
